core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -134,15 +125,28 @@
    * @param otherFields other fields to index the set
    */
   @SafeVarargs
-  public IndexedSet(FieldIndex<T> field, FieldIndex<T>... otherFields) {
-    Map<FieldIndex<T>, ConcurrentHashMap<Object, ConcurrentHashSet<T>>> indexMap =
-        new HashMap<>(otherFields.length + 1);
-    indexMap.put(field, new ConcurrentHashMap<Object, ConcurrentHashSet<T>>(8, 0.95f, 8));
-    for (FieldIndex<T> fieldIndex : otherFields) {
-      indexMap.put(fieldIndex, new ConcurrentHashMap<Object, ConcurrentHashSet<T>>(8, 0.95f, 8));
+  public IndexedSet(IndexDefinition<T> field, IndexDefinition<T>... otherFields) {
+    // count the numbers of two index types
+    Iterable<IndexDefinition<T>> fields =
```

Comment:
I would probably prefer to call this `indexDefinitions` or something. I think `fields` is misleading.